---
title: "Distinguishing SNPs for DFTD1 and DFTD1 lines"
author: "Mark Ziemann"
date: "`r Sys.Date()`"
output:
  html_document:
    toc: true
    toc_float: true
    code_folding: hide
    fig_width: 7
    fig_height: 7
theme: cosmo
---

Source: https://github.com/markziemann/dftd

## Introduction

Here I'm reading in some SNP files from the cancer cell article: 
https://www.cell.com/cancer-cell/fulltext/S1535-6108(18)30117-X
to identify SNPs in exons that could be used to distinguish the lines.

The two files are SNPs which are present in the DFTD1 and DFTD2 cell types respectively.
Although I'm not sure if these are mutually exclusive, the SNP sets may have an overlap,
which I will test.

There are 8 sample which are of interest to us.

| Sample type | Sample label |
| --- | --- |
| DFT1 | 86T |
| DFT1 | 88T |
| DFT2 | 202T2 |
| DFT2 | 203T3 |
| Normal | 91H |
| Normal | 31H |
| Normal | 202H1 |
| Normal | 203H |


```{r,packages}

suppressPackageStartupMessages({
  library("eulerr")
  library("kableExtra")
  library("GenomicRanges")
  library("readxl")
})

```


## Load data

Here I'm loading the data in and only selecting the DFTD lines.

```{r,load1}

d1 <- read.table("DFT1_SNVs.txt",header=TRUE)
head(d1)
str(d1)

# subset to obtain DFTD lines only
d1 <- d1[,1:8]
head(d1)

```

## Cleaning data

Now look at the SNPS which are common to the two DFTD1 lines 86 and 88

First need to parse out the data into numerical format.

```{r,dftd1}

g1 <- d1[,5:ncol(d1)]
x86 <- do.call(rbind,strsplit(g1$X86T,"/"))
x88 <- do.call(rbind,strsplit(g1$X88T,"/"))
x202 <- do.call(rbind,strsplit(g1$X202T2,"/"))
x203 <- do.call(rbind,strsplit(g1$X203T3,"/"))

g1 <- cbind(x86,x88,x202,x203)
g1 <- apply(g1,2,as.numeric)

colnames(g1) <- c("x86_alt","x86_total",
  "x88_alt","x88_total",
  "x202_alt","x202_total",
  "x203_alt","x203_total")
head(g1)

x1 <- cbind(d1,g1)
str(x1)

```

## Filtering

Now we can filter further.

Use a 25% allele frequency filter to identify the SNPs for each DFTD line.

```{r,filter1}

x86 <- x1[(which(x1$x86_alt / x1$x86_total > 0.25)),]
x88 <- x1[(which(x1$x88_alt / x1$x88_total > 0.25)),]
x202 <- x1[(which(x1$x202_alt / x1$x202_total > 0.25)),]
x203 <- x1[(which(x1$x203_alt / x1$x203_total > 0.25)),]
str(x86)
str(x88)
str(x202)
str(x203)

```

Importantly, there were approx 6000 SNPs that could identify DFTD1 lines but only 7 which could 
identify DFTD2 lines.

Let's see whether these are specific to one cell line.

```{r,filter2}

x86_2 <- x1[which(x1$x86_alt / x1$x86_total > 0.25 & x1$x202_alt / x1$x202_total < 0.25 &  x1$x203_alt / x1$x203_total < 0.25 ),]
dim(x86)
dim(x86_2)

x88_2 <- x1[which(x1$x88_alt / x1$x88_total > 0.25 & x1$x202_alt / x1$x202_total < 0.25 &  x1$x203_alt / x1$x203_total < 0.25 ),]
dim(x88)
dim(x88_2)

```

This shows that of the 5678 SNPs specific to x86, 5667 were specific to this line and not present in x202 or s203.

Of the 6231 SNPs specific to x88, 6222 were specific to this line and not present in x202 or s203.

This means that 5000-6000 SNPs can be used to distinguish genomes.

```{r,filter3}

d1_snps <- x1[which(x1$x86_alt / x1$x86_total > 0.25 & x1$x88_alt / x1$x88_total > 0.25),]
str(d1_snps)

d2_snps <- x1[which(x1$x202_alt / x1$x202_total > 0.25 & x1$x203_alt / x1$x203_total > 0.25),]
str(d2_snps)

```

This indicates 2745 SNPs which are common to the DFTD1 cell lines, while only 3 SNPs were common to
DFTD2 cell lines.

Given that there are are only 2745 which are common to DFTD1 lines, we better focus on the 6222
which can distinguish x88.

## Exonic SNPs

Next, lets look at the 6222 SNPs that are present in x88 and not in the DFTD2 lines and see how many are located 
in exons.

The exon bedfile was generated with the gff2bed.sh script. 
That script extracts CDS coordinates the GTF file.
We use the GTF as input because it has the gene ID number which can be matched to the RNA-seq.

```{r,exons1}

cds <- read.table("../gene_annotation/Sarcophilus_harrisii.DEVIL7.0.90.gtf.cds.bed")
colnames(cds) <- c("seqname","start","end","gene_name")

# GRanges()
cdsr <- GRanges(seqnames=cds$seqname,ranges=IRanges(start=cds$start, end=cds$end) )
cdsr$gene_name <- cds$gene_name
head(cdsr)

```

## SNP Exon Overlap

The problem is that exons have this sort of seq name (GL834412.1) while the variants have this type 
(Chr1_supercontig_000000000)

The correspondence map of these naming conventions is given here: https://plos.figshare.com/articles/dataset/Genome_coordinates_/13283417/1

```{r,map1}

head(x88_2)

map_tbl <- read.table("../gene_annotation/supercontig2gl.tsv",header=TRUE)
head(map_tbl)
x88_2$SCAFFOLD <- map_tbl[match(x88_2$SCAFFOLD,map_tbl$TCG_ID),"ENSEMBL_ID"]
head(x88_2)

```

Now we can intersect

```{r,intersect}

x88_r <- GRanges(seqnames=x88_2$SCAFFOLD, ranges=IRanges(start=x88_2$POS, end=x88_2$POS+1) )

x88_r

ol <- findOverlaps(x88_r,cdsr)

x88_r[queryHits(ol)]

cds_vars <- as.data.frame(cdsr[subjectHits(ol)])

cds_vars %>% 
  kbl(caption="x88 vars in protein coding genes") %>% 
  kable_paper("hover", full_width = F)

length(unique(cds_vars$gene_name))

```

This result shows that from the 6222 variants that can distinguish x88, only 50 are in
protein-coding genes (0.08%).

Each of these genes contained only 1 variant, so there are only 50 genes which can be
distinguished.

This is consistent with only 1.1% of the genome being in expressed exons.
And that exons are anticipated to accumulate variants at a rate slower than non-exonic regions
due to their essential function.

## Which genes are expressed

From these 50 genes, not all of them will be expressed in the cells.
We can look at the list of genes which are expressed in RNA-seq to give a realistic 
propostion of how many can be distinguished by RNA-seq.

```{r,rnaseq}

rnaseq <- readxl::read_xlsx("../oncotarget-09-15895-s002.xlsx",col_names=TRUE,skip=1)
rnaseq <- as.data.frame(rnaseq)
rnaseq_expressed <- rnaseq[which(rowMeans(rnaseq[,grep("counts",colnames(rnaseq))])>10),]
dim(rnaseq_expressed)

rnaseq_expressed$gene_id 

cds_vars$gene_name %in% rnaseq_expressed$gene_id
cds_vars$gene_name[which(cds_vars$gene_name %in% rnaseq_expressed$gene_id)]
length(cds_vars$gene_name[which(cds_vars$gene_name %in% rnaseq_expressed$gene_id)])

```

## Session information

```{r,sessioninfo}

sessionInfo()

```
